cmd/google_cloud_ops_agent_engine/main.go

```diff
@@ -41,8 +44,34 @@ func main() {
 func run() error {
 	// TODO(lingshi) Move this to a shared place across Linux and Windows.
 	confDebugFolder := filepath.Join(os.Getenv("RUNTIME_DIRECTORY"), "conf", "debug")
-	if err := confgenerator.MergeConfFiles(*input, confDebugFolder, "linux", apps.BuiltInConfStructs); err != nil {
+	mergedConfig, err := confgenerator.MergeConfFiles(*input, confDebugFolder, "linux", apps.BuiltInConfStructs)
```

Comment:
In the interest of minimizing this PR's diff, could we just have `MergeConfFiles` return the built-in config bytes as well?

I also think we should refactor this code (outside of this PR) to simplify the flow. Maybe something like:

1. Read input files
2. Generate outputs
3. Log inputs + outputs
4. Write output files

@quentinmit might have other ideas too.

Reply:
4 is unnecessary, the only reason we write the intermediate files now is because of the current structure of the code. 1-3 is sufficient for this feature of logging the config.

Reply:
Have we decided that we don't want to keep those files at all? In the event that Fluent Bit fails for whatever reason, we won't be able to see the logs in Cloud Logging and having these files locally for customers to send over and/or look at themselves could be pretty useful. I kept it here just for debuggability reasons but happy to remove it if we decide.

Reply:
The configs will still end up in the logs on the local disk without 4, so it doesn't affect the "Fluent Bit fails" case.

Reply:
Refactored to not write these output files and merge in-memory.